cqrs-codegen/impl/src/aggregate.rs

```diff
@@ -0,0 +1,193 @@
+
+    let const_val = parse_aggregate_type(&meta)?;
+    let const_doc = format!("Type name of [`{}`] aggregate", input.ident);
+    let additional = quote! {
+        #[doc = #const_doc]
+        pub const AGGREGATE_TYPE: ::cqrs::AggregateType = #const_val;
+    };
+
+    let data = match &input.data {
+        syn::Data::Struct(data) => data,
+        _ => unreachable!(),
+    };
+
+    let (id_type, id_field) = get_id(&data.fields)?;
+
+    let body = quote! {
+        type Id = #id_type;
+
+        #[inline(always)]
+        fn aggregate_type(&self) -> ::cqrs::AggregateType {
+            Self::AGGREGATE_TYPE
+        }
+
+        #[inline(always)]
+        fn id(&self) -> &Self::Id {
+            &self.#id_field
+        }
+    };
+
+    util::render_struct(
+        &input,
+        quote!(::cqrs::Aggregate),
+        body,
+        Some(additional),
+    )
+}
+
+/// Reports error if [`crate::aggregate_derive`] macro applied to enums.
+fn derive_enum(input: syn::DeriveInput) -> Result<TokenStream> {
+    match input.data {
+        syn::Data::Enum(data) => Err(Error::new(
+            data.enum_token.span(),
+            format!("Structs are not supported for deriving {}", TRAIT_NAME),
+        )),
+        _ => unreachable!(),
+    }
+}
+
+/// Parses type of [`cqrs::Aggregate`] from `#[aggregate(...)]` attribute.
+fn parse_aggregate_type(meta: &util::Meta) -> Result<String> {
+    let lit: &syn::LitStr = util::parse_lit(
+        meta,
+        "type",
+        &["type"],
+        ATTR_NAME,
+        "= \"...\"",
+    )?;
+
+    Ok(lit.value())
+}
+
+/// Infers-or-finds-via-attribute an `id` field of the aggregate.
+fn get_id(fields: &syn::Fields) -> Result<(&syn::Type, TokenStream)> {
+    let mut id = None;
+
+    for (index, field) in fields.iter().enumerate() {
+        let meta = util::find_nested_meta(&field.attrs, ATTR_NAME)?;
+
+        let meta = match meta {
+            Some(meta) => meta,
+            None => continue,
+        };
+
+        let explicit = util::parse_flag(
+            &meta,
+            "id",
+            &["id"],
+            ATTR_NAME,
+        )?;
+
+        if explicit {
+            let span = field.span();
+            if id.replace((index, field)).is_some() {
+                return Err(Error::new(
+                    span,
+                    "Multiple fields marked with '#[aggregate(id)]' attribute; \
+                    only single '#[aggregate(id)]' attribute allowed per struct",
+                ));
+            }
+        }
+    }
+
+    let named = match fields {
+        syn::Fields::Named(_) => true,
+        _ => false,
+    };
+
+    if id.is_none() && named {
+        id = fields
+            .iter()
+            .enumerate()
+            .find(|(_, field)| {
+                match &field.ident {
+                    Some(ident) => ident == "id",
+                    None => false,
+                }
+            });
+    }
+
+    id
+        .map(|(index, field)| {
```

Comment:
You definitely should apply `cargo +nightly fmt -p cqrs-codegen-impl` before requesting review.